api/v1alpha1/drpolicy_types.go

```diff
@@ -25,8 +25,20 @@ type DRPolicySpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
+	// schedule is a cronspec (https://en.wikipedia.org/wiki/Cron#Overview) that
+	// can be used to schedule replication to occur at regular, time-based
+	// intervals.
+	//+kubebuilder:validation:Pattern=`^(\d+|\*)(/\d+)?(\s+(\d+|\*)(/\d+)?){4}$`
+	Schedule *string `json:"schedule"`
```

Comment:
The ceph-csi [schedule](https://github.com/ceph/ceph-csi/blob/2782878ea20c7d49f392ccdb948001eb0e1b83e0/internal/rbd/replicationcontrollerserver.go#L64-L68) is not a cron spec, are we converting this from a cron spec to a schedule as in CSI to find the VolRepClass? If not we may just want to stick to the CSI format for now.

Reply:
ACK. Couple of questions.

1) Does this mean, from DRPolicy spec (i.e. higher layer) itself we mandate the schedule to be given in CSI format? Or do you expect the cron spec to be converted to CSI format at the lower layer like VRG(or wherever necessary, such drpc/DrPolicy while validating the schedule etc)?

2) Is the schedule format you suggested specific to ceph-csi driver or is it the schedule format used in CSI itself (i.e. any csi driver will have the same format)?

Reply:
The schedule is a VolRepClass `parameter` which hence can be different for different providers.

I would suggest we keep it simple for now and do (1) above, i.e take a snapshotInterval in the same format, and compare it to decide the VolRepClass as needed. As we only have a single implementation of the same.